VMEncryption/main/Utils/HandlerUtil.py

```diff
@@ -403,72 +408,108 @@ def redo_current_status(self):
+        sys.exit(0)
+
     def do_status_report(self, operation, status, status_code, message):
-        latest_seq_num = self.get_latest_seq()
-        if (latest_seq_num >= 0): 
-            latest_seq = str(self.get_latest_seq())
-        else:
-            self.log("sequence number could not be derived from settings files, using 0.status")
-            latest_seq = "0"
-
-        self._context._status_file = os.path.join(self._context._status_dir, latest_seq + '.status')
-
-        if message is None:
-            message = ""
-
-        message = filter(lambda c: c in string.printable, message)
-        message = message.encode('ascii', 'ignore')
-
-        self.log("[StatusReport ({0})] op: {1}".format(latest_seq, operation))
-        self.log("[StatusReport ({0})] status: {1}".format(latest_seq, status))
-        self.log("[StatusReport ({0})] code: {1}".format(latest_seq, status_code))
-        self.log("[StatusReport ({0})] msg: {1}".format(latest_seq, message))
-
-        tstamp = time.strftime(DateTimeFormat, time.gmtime())
-        stat = [{
-            "version" : self._context._version,
-            "timestampUTC" : tstamp,
-            "status" : {
-                "name" : self._context._name,
-                "operation" : operation,
-                "status" : status,
-                "code" : status_code,
-                "formattedMessage" : {
-                    "lang" : "en-US",
-                    "message" : message
-                }
-            }
-        }]
-
-        if self.disk_util:
-            encryption_status = self.disk_util.get_encryption_status()
-
-            self.log("[StatusReport ({0})] substatus: {1}".format(latest_seq, encryption_status))
-
-            substat = [{
-                "name" : self._context._name,
-                "operation" : operation,
-                "status" : status,
-                "code" : status_code,
-                "formattedMessage" : {
-                    "lang" : "en-US",
-                    "message" : encryption_status
+        lock = None
+        try:
+            if not os.path.exists(self.config_folder):
+                os.makedirs(self.config_folder)
+            lock = ProcessLock(self, self.status_lock_file_path)
+            while not lock.try_lock():
+                self.log("Cannot acquire lock to write status.")
+                time.sleep(0.1)
+
+            latest_seq_num = self.get_latest_seq()
+            if (latest_seq_num >= 0): 
+                latest_seq = str(self.get_latest_seq())
+            else:
+                self.log("sequence number could not be derived from settings files, using 0.status")
+                latest_seq = "0"
+
+            self._context._status_file = os.path.join(self._context._status_dir, latest_seq + '.status')
+            if message is None:
+                message = ""
+                
+            message = filter(lambda c: c in string.printable, message)
```

Comment:
Is this valid lambda syntax on Python 2.6?

Reply:
This syntax is already part of current extension but I will double check it.

Reply:
feel free to close, if this syntax is already in use then it's a moot point